orderer/common/cluster/clusterservice.go

```diff
@@ -85,14 +85,21 @@ func (s *ClusterService) Step(stream orderer.ClusterNodeService_StepServer) erro
 		return err
 	}
 
-	authReq, configUpdatedTime, err := s.VerifyAuthRequest(stream, request)
+	authReq, err := s.VerifyAuthRequest(stream, request)
 	if err != nil {
-		s.Logger.Warnf("service authentication failed with error: %v", err)
+		s.Logger.Warnf("service authentication of %s failed with error: %v", addr, err)
 		return status.Errorf(codes.Unauthenticated, "access denied")
 	}
 
+	defer s.Logger.Debugf("Closing connection from %s(%s)", commonName, addr)
+
+	s.Lock.Lock()
```

Comment:
why are you protecting this with a lock? this is an atomic operation and it's memory coherent. 

This is also why I put a sync.map in the first place, so we could do this under a read lock. 

You can move the RLock from `VerifyAuthRequest` to line 87 to 100.

Reply:
thought inside the read lock, updating/writing into the map may not be correct semantically.  Now. I updated it.

Reply:
We always write different keys since the key is atomically retrieved and is incremented atomically.

Since the map is thread safe, we can do it concurrently.